stats/Record.md

```diff
@@ -64,16 +67,33 @@ argument. e.g. `record(List<Measurement>)` or `record(...Measurement)`.
 Example in Java
 
 ```java
+// Static constants.
 private static final MeasureDouble RPC_LATENCY =
     MeasureDouble.create("grpc.io/client/latency", "latency", "ms");
 private static final MeasureLong RPC_BYTES_SENT =
     MeasureLong.create("grpc.io/client/bytes_sent", "bytes sent", "kb");
+private static final TagKey MY_KEY = TagKey.create("my.org/key");
+```
 
+```java
+// Record against implicit context.
 MeasurementMap measurementMap = new MeasurementMap();
 measurementMap.put(RPC_LATENCY, 10.3);
 measurementMap.put(RPC_BYTES_SENT, 124);
 measurementMap.record();  // reads context from thread-local.
+```
 
+```java
+// Record against explicit context.
+MeasurementMap measurementMap = new MeasurementMap();
+measurementMap.put(RPC_LATENCY, 15);
+measurementMap.put(RPC_BYTES_SENT, 200);
+TagValue value = TagValue.create("some value");
+measurementMap.record(
+    Tags.getTagger().currentBuilder().put(MY_KEY, value).build());  // record against an extra tag.
```

Comment:
I think this is exactly the same problem as Go. what we want in that is something like:
private static final Map<TagKey, TagValue> extraTags = ...
measurementMap.recordWithTags(currentTagContext, extraTags)

Reply:
What is the benefit of the API that takes a Map?  It looks like it would have the same behavior as the example in this PR, except that it doesn't allow the library to skip creating the map as an optimization when OpenCensus isn't enabled.

Reply:
Please comment on the main issue.